infrastructure/datastore/db.go

```diff
@@ -0,0 +1,74 @@
+package datastore
+
+import (
+	"encoding/csv"
+	"io"
+	"log"
+	"os"
+	"strconv"	
+
+	"api-booking-time/domain/model"
+	"api-booking-time/usecase/repository"
+	ir "api-booking-time/interface/repository"
+)
+
+type DbRepository struct {
+	Centres interface{ repository.CentreRepository }
+}
+
+func OpenDb() *DbRepository {
+	centres := initialize()
+
+	return &DbRepository{Centres: centres}
+}
+
+func initialize() repository.CentreRepository {
+	filename := "./assets/centres.csv"
+	file, err := os.Open(filename)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	defer file.Close()
+	return ir.OpenCentreRepository(loadData(file))
+}
+
+func loadData(r io.Reader) *[]*model.Centre {
+	reader := csv.NewReader(r)
+
+	ret := make([]*model.Centre, 0, 0)
+
+	for {
+		row, err := reader.Read()
+		if err == io.EOF {
+			log.Println("End of File")
+			break
+		} else if err != nil {
+			log.Println(err)
+			break
+		}
```

Comment:
Worst case scenario, you check twice for an error only once, but the way you had it, every line you make two error checks.
```suggestion
		if err != nil {
			if err == io.EOF {
				log.Println("End of File")
				break
			}
			log.Println(err)
			break
		}
```